enumeratum-core/src/test/scala/enumeratum/Models.scala

```diff
@@ -309,3 +305,16 @@ object InTheWoods {
 
   }
 }
+
+object Inheritance {
+  sealed trait Word extends EnumEntry
+
+  object Word extends Enum[Word] {
+    sealed class Greeting extends Word
```

Comment:
If you remove `sealed` from `class Greeting`, the test will fail

Reply:
@cchantep  Notice that `class Derived` is not sealed in my example.

To reproduce the issue-1, you should remove `sealed` from `class Greeting`.

Reply:
Not having `sealed` cannot (and should not) be supported

Reply:
It works perfectly well with scala-2.13 and enumeraturm 1.7.0 as shown in this gist:

```txt
$ scala-cli https://gist.github.com/mushtaq/5055834b968bf17c84bfa28f36980924
Compiling project (Scala 2.13.10, JVM)
Compiled project (Scala 2.13.10, JVM)
Vector(E1, E2)
```

Note that `Base` is still `sealed`.

`Derived` does not require to be sealed, because we are creating an `Enum[Base]` and not an `Enum[Derived]`

Reply:
@mushtaq That it works with 2.13 doesn't give it more sense.

- You may be lucky that Scala 2.13 find subobjects, but it has always add limitation for the macros to find subtypes.
- Such modeling looks like an edge one for me that I would not encourage, and even prevent and detect as erroneous.
- I won't take more time on such case, so feel free to block this PR and close it.

Reply:
@cchantep I would love this PR to be merged ASAP because it is so useful to have enumeratum on scala-3.

The issues pointed out here are not blockers because I already have found workarounds.

I am sharing them just so that you or @lloydmeta may want to make use of them!

Thanks a lot once again for your responsiveness 👍